### PR TITLE
add series override option to hide tooltip

### LINKED ISSUE #3341: Series override: Hide / Show Only in Tooltip
Just upgraded to 2.5, and discovered "legend: false" -- this is great! It would be nice if there was also "tooltip: false" so that items could be excluded from the tooltip, as well as the legend.

There are cases where the query returns values that are necessary for computation, but distract from the graph. In these cases it is useful to have both "legend: false" and "tooltip: false," so that the computation can be performed, but not distract from the visual representation.

### PR DESCRIPTION
**Pull request by mtanda**
For drawing dynamic threshold line, I want to use this.

Fix https://github.com/grafana/grafana/issues/3341

### CODE REVIEW

**Comment by marefr** [comment]
Could you please rebase this and we'll have a look at it. Thanks

**Comment by mtanda** [comment]
rebased! Please check!!

File: public/app/plugins/panel/graph/series_overrides_ctrl.ts [hunk 1]
```diff
@@ -152,6 +152,7 @@ export function SeriesOverridesCtrl($scope, $element, popoverSrv) {
   $scope.addOverrideOption('Z-index', 'zindex', [-3, -2, -1, 0, 1, 2, 3]);
   $scope.addOverrideOption('Transform', 'transform', ['negative-Y']);
   $scope.addOverrideOption('Legend', 'legend', [true, false]);
+  $scope.addOverrideOption('Tooltip', 'tooltip', [true, false]);
```
Comment (torkelo) [inline public/app/plugins/panel/graph/series_overrides_ctrl.ts]: I wonder if inverting this makes more sense and is more intutive, make the option read: `Hide in tooltip` where default is false. Boolean override options that default to false are better I think.

**Review by torkelo — CHANGES_REQUESTED** [review]

File: public/app/plugins/panel/graph/series_overrides_ctrl.ts (hunk 1 above)
Comment (mtanda) [reply public/app/plugins/panel/graph/series_overrides_ctrl.ts]: It is more intuitive. I changed like `hideTooltip`.

File: public/app/core/time_series2.ts [hunk 2]
```diff
@@ -107,6 +108,7 @@ export default class TimeSeries {
     this.valueFormater = kbn.valueFormats.none;
     this.stats = {};
     this.legend = true;
+    this.hideTooltip = false;
```
Comment (torkelo) [inline public/app/core/time_series2.ts]: not sure boolean instance variables needs to be initialized to false

Comment (mtanda) [reply public/app/core/time_series2.ts]: without initialization, the value is `undefined`.
I like boolean initialized value, but other boolean variables are not initialized, so I remove the initialization.

**Comment by torkelo** [comment]
thanks @mtanda , sorry for taking way to long to merge this.

**Comment by mtanda** [comment]
thanks :-)

**Comment by Sandwichzhao** [comment]
why I can not find hideTooltip false option. mysql grafana 5.4.0-pre1,could you tell me why？
image

**Comment by marefr** [comment]
@Sandwichzhao it's in series override:

image

**Comment by Sandwichzhao** [comment]
that is terrible,I can't find it.
image
image
Is it because my version is too low?

**Comment by Sandwichzhao** [comment]
oh yeah, my guess is right .The plugin is too low.When I update all plugins ， it's work.Thanks for your help!!

**Pull request merged**